app/javascript/packs/flagUserModal.js

```diff
@@ -0,0 +1,158 @@
+import { request } from '../utilities/http';
+
+const modalSnackbarHTML = `
+<div class="crayons-snackbar">
+  <div class="crayons-snackbar__item" id="vomit-all-snackbar">
+    <div class="crayons-snackbar__body">
+      <p>All posts by this author will be less visible.</p>
+    </div>
+  </div>
+</div>
+`;
+
+const toggleFlagUserModal = () => {
+  const modalContainer = document.querySelector('.flag-user-modal-container');
+  modalContainer.classList.toggle('hidden');
+
+  if (!modalContainer.classList.contains('hidden')) {
+    window.scrollTo(0, 0);
+    document.querySelector('body').style.height = '100vh';
+    document.querySelector('body').style.overflowY = 'hidden';
+  } else {
+    document.querySelector('body').style.height = 'inherit';
+    document.querySelector('body').style.overflowY = 'inherit';
+  }
+};
+
+const flashSnackbar = (snackbar) => {
+  setTimeout(() => {
+    snackbar.classList.remove('flex');
+  }, 3000);
+  snackbar.classList.add('flex');
+};
+
+export function initializeFlagUserModal() {
+  // eslint-disable-next-line no-undef
+  const user = userData();
+  const {
+    authorId: articleAuthorId,
+    authorClassName: articleAuthorClassName,
+  } = document.getElementById('article-show-container').dataset;
+
+  const flagUserModalHTML = `
```

Comment:
In a future PR, it would be nice to use the Preact `<Modal />` component instead, but at the moment, it is still under review.

Reply:
Ok!